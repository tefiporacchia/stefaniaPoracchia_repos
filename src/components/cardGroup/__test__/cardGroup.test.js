import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import CardGroup from "../cardGroup";

describe('Card Group Tests', () => {

    let cards;

    beforeAll( () => {

        cards = [
            {
                name:'charizard',
                sprites: {
                    other: {
                        dream_world: {
                            front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/2.svg',
                        }
                    }
                },
                id:1
            },
            {
                name:'woopy',
                sprites: {
                    other: {
                        dream_world: {
                            front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/43.png',
                        }
                    }
                },
                id:2
            },
            {
                name:'daisy',
                sprites: {
                    other: {
                        dream_world: {
                            front_default: 'ttps://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/39.png',
                        }
                    }
                },
                id:3
            },
            {
                name:'snow',
                sprites: {
                    other: {
                        dream_world: {
                            front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/39.png',
                        }
                    }
                },
                id:4
            }]
    });

    test('renders correctly', () => {
        const tree = renderer
            .create(<CardGroup elems={cards}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("must display all ids", () => {
        render(<CardGroup elems={cards}/>);
        expect(screen.getByText("#01")).toBeInTheDocument();
        expect(screen.getByText("#02")).toBeInTheDocument();
        expect(screen.getByText("#03")).toBeInTheDocument();
        expect(screen.getByText("#04")).toBeInTheDocument();
    })

    test("must display all pictures", () => {
        render(<CardGroup elems={cards}/>);
        const img1 = screen.getAllByRole("img");
        expect(img1[0]).toHaveAttribute('src', cards[0].img);
        expect(img1[0]).toHaveAttribute('alt', 'charizard');
        expect(img1[1]).toHaveAttribute('src', cards[1].img);
        expect(img1[1]).toHaveAttribute('alt', 'woopy');
        expect(img1[2]).toHaveAttribute('src', cards[2].img);
        expect(img1[2]).toHaveAttribute('alt', 'daisy');
        expect(img1[3]).toHaveAttribute('src', cards[3].img);
        expect(img1[3]).toHaveAttribute('alt', 'snow');
    })

    test("must display the pokemons names", () => {
        render(<CardGroup elems={cards}/>);
        expect(screen.getByText("Charizard")).toBeInTheDocument();
        expect(screen.getByText("Woopy")).toBeInTheDocument();
        expect(screen.getByText("Daisy")).toBeInTheDocument();
        expect(screen.getByText("Snow")).toBeInTheDocument();
    })


})