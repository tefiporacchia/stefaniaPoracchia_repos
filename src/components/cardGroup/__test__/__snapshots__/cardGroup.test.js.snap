// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Group Tests renders correctly 1`] = `
<div
  className="card-group"
>
  <div
    className="card-container number-1"
    onClick={[Function]}
  >
    <img
      alt="charizard"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/2.svg"
    />
    <div
      className="number"
    >
      <small>
        #0
        1
      </small>
    </div>
    <h3>
      Charizard
    </h3>
  </div>
  <div
    className="card-container number-2"
    onClick={[Function]}
  >
    <img
      alt="woopy"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/43.png"
    />
    <div
      className="number"
    >
      <small>
        #0
        2
      </small>
    </div>
    <h3>
      Woopy
    </h3>
  </div>
  <div
    className="card-container number-3"
    onClick={[Function]}
  >
    <img
      alt="daisy"
      src="ttps://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/39.png"
    />
    <div
      className="number"
    >
      <small>
        #0
        3
      </small>
    </div>
    <h3>
      Daisy
    </h3>
  </div>
  <div
    className="card-container number-0"
    onClick={[Function]}
  >
    <img
      alt="snow"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/39.png"
    />
    <div
      className="number"
    >
      <small>
        #0
        4
      </small>
    </div>
    <h3>
      Snow
    </h3>
  </div>
</div>
`;

exports[`Main page mount renders correctly 1`] = `
<div
  className="card-group"
>
  <div
    className="card-container number-1"
    onClick={[Function]}
  >
    <img
      alt="charizard"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/2.svg"
    />
    <div
      className="number"
    >
      <small>
        #0
        1
      </small>
    </div>
    <h3>
      Charizard
    </h3>
  </div>
  <div
    className="card-container number-2"
    onClick={[Function]}
  >
    <img
      alt="woopy"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/2.svg"
    />
    <div
      className="number"
    >
      <small>
        #0
        2
      </small>
    </div>
    <h3>
      Woopy
    </h3>
  </div>
  <div
    className="card-container number-3"
    onClick={[Function]}
  >
    <img
      alt="daisy"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/2.svg"
    />
    <div
      className="number"
    >
      <small>
        #0
        3
      </small>
    </div>
    <h3>
      Daisy
    </h3>
  </div>
  <div
    className="card-container number-0"
    onClick={[Function]}
  >
    <img
      alt="snow"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/2.svg"
    />
    <div
      className="number"
    >
      <small>
        #0
        4
      </small>
    </div>
    <h3>
      Snow
    </h3>
  </div>
</div>
`;
